src/modules/returns/lib/csv-adapter/date-parser.js

```diff
@@ -51,7 +52,8 @@ const getDateFrequency = date => {
     const m = parsers[timePeriod](date)
     return m.isValid() ? { timePeriod, moment: m } : null
   })
-  return compact(result)[0]
+
+  return result.filter(x => !!x)[0]
```

Comment:
If I'm reading this correctly we're populating `result` with either a parsed date or `null`, then returning the first truthy value (ie. the first successful attempt at parsing). So instead of filtering out the falsey values then returning the first item in the array, we could use this to just return the first truthy value:
```suggestion
  return result.find(x => x)
```